<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jennifer Blanc" />


<title>Selection on expression of coexpression clusters</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Blancetal</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jgblanc/Blancetal">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Selection on expression of coexpression clusters</h1>
<h4 class="author">Jennifer Blanc</h4>
<h4 class="date">3/13/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Blancetal/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200217code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200217)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200217code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200217)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjgblancBlancetaltree3294a81f9ac4df07e1236449b04b827bf1d80077targetblank3294a81a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jgblanc/Blancetal/tree/3294a81f9ac4df07e1236449b04b827bf1d80077" target="_blank">3294a81</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjgblancBlancetaltree3294a81f9ac4df07e1236449b04b827bf1d80077targetblank3294a81a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jgblanc/Blancetal/tree/3294a81f9ac4df07e1236449b04b827bf1d80077" target="_blank">3294a81</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/quaint-results.rda
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  data/schaefer_clusters.csv~
    Untracked:  figures/Supplement_Ve.png
    Untracked:  figures/phist.png

Unstaged changes:
    Modified:   analysis/Environmental_variance.Rmd
    Modified:   analysis/Selection_on_Expression_of_Env_Rsponse_Genes.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Selection_on_expression_of_coexpreession_clusters.Rmd</code>) and HTML (<code>docs/Selection_on_expression_of_coexpreession_clusters.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/3294a81f9ac4df07e1236449b04b827bf1d80077/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">3294a81</a>
</td>
<td>
jgblanc
</td>
<td>
2020-09-30
</td>
<td>
new clusters
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgblanc/Blancetal/9ede1d414a3847f17a17fcb9d80d2498d3638f7d/docs/Selection_on_expression_of_coexpreession_clusters.html" target="_blank">9ede1d4</a>
</td>
<td>
jgblanc
</td>
<td>
2020-04-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/69c5a296f43b756db20e5b6ff4bbfa95cebaa930/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">69c5a29</a>
</td>
<td>
jgblanc
</td>
<td>
2020-04-24
</td>
<td>
Finished clusters
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgblanc/Blancetal/e26e75465c0a76c1880ad6e2502b916761f92d9c/docs/Selection_on_expression_of_coexpreession_clusters.html" target="_blank">e26e754</a>
</td>
<td>
jgblanc
</td>
<td>
2020-04-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/3a6666a0f5cccdbfe147cf6d8a158ebfcc72d670/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">3a6666a</a>
</td>
<td>
jgblanc
</td>
<td>
2020-04-24
</td>
<td>
Finished clusters
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/8298d4dbcb85647c6a5a3e38ddd61fb4eb8cd84e/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">8298d4d</a>
</td>
<td>
GitHub
</td>
<td>
2020-04-16
</td>
<td>
Merge branch ‘master’ into master
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/6b00f4790b7b9dc03222e98662c234e496d5b6a1/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">6b00f47</a>
</td>
<td>
em
</td>
<td>
2020-03-27
</td>
<td>
stuff
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/a63fa862c7aa1b8aec99a8ef9299072467349752/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">a63fa86</a>
</td>
<td>
em
</td>
<td>
2020-03-26
</td>
<td>
stuff
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgblanc/Blancetal/7591f0dc89d964a3a09cf5200bc7819552ca2b80/docs/Selection_on_expression_of_coexpreession_clusters.html" target="_blank">7591f0d</a>
</td>
<td>
jgblanc
</td>
<td>
2020-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/df4fe40063954a4423f40f617a94fbbe5cb3842d/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">df4fe40</a>
</td>
<td>
jgblanc
</td>
<td>
2020-03-16
</td>
<td>
added more pvals
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/2cb89cd06d2ebc12c3eb3103c6b0c5b78c690ba8/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">2cb89cd</a>
</td>
<td>
jgblanc
</td>
<td>
2020-03-13
</td>
<td>
adding coexpression stuff
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Here is the code to test for expression on coeexpression clusters from Walleey et al.</p>
</div>
<div id="code" class="section level2">
<h2>Code</h2>
<p>For this analysis we want to test for selection within specific coexpression modules. We used coexpression modules from Walley et al. (2016) who used weight gene coexpression network analysis (WGCNA) to to group together genes that were similarly expressed in at least 4 tissues in one maize inbred line. Their analysis resulted in 66 co-expression networks. Below we will load their co-expression networks and select all the clusters that have at least 100 genes in them.</p>
<pre class="r"><code>modules &lt;- read.delim(&quot;../data/Modules.txt&quot;,na.strings=c(&quot;&quot;,&quot;NA&quot;), stringsAsFactors = F)
num_genes &lt;- apply(modules, 2, function(x) length(which(!is.na(x))))
num_genes_100 &lt;- which(num_genes &gt;= 100)
modules  &lt;- modules[,num_genes_100]</code></pre>
<p>We now have 51 modules that have at least 100 genes. We will now get the median expression of all the genes that are present in our RNAseq dataset for each module and then test for selection on the median values in each cluster. The code below is similar code used to identify selected genes, we are simply conducting the test on median expression values in each line instead of a single gene expression in each line. Again we test for selection on the first 5 PCs and use the last half of PCs to estimate <span class="math inline">\(V_a\)</span>. Th function will return the p-values from the <span class="math inline">\(Q_{pc}\)</span> test for all tissues.</p>
<pre class="r"><code>cluster_analysis_func &lt;- function(modules) {

alltissues = c(&#39;GRoot&#39;,&quot;Kern&quot;,&quot;LMAD26&quot;,&quot;LMAN26&quot;, &quot;L3Tip&quot;,&quot;GShoot&quot;,&quot;L3Base&quot;)

alltissuemodules = lapply(alltissues, function(mytissue){
  
##get the names of genes we have expression data for in each tissue
exp &lt;- read.table(paste(&quot;../data/Raw_expression/&quot;,mytissue,&quot;.txt&quot;, sep=&quot;&quot;)) #  read in expression level
geneNames = names(exp)

#get the median expression level for each module in this tissue
moduleExpression = apply(modules, 2, function(x){
  olap = x[x %in% geneNames] #get genes in the module tha we have expression data for
  moduleExp = dplyr::select(exp, all_of(olap)) #pull out the expression data for these genes
  medianExp = apply(moduleExp, 1, median)
  return(medianExp)
})

####identify selection in each of these modules
# Read in tissue specific kinship matrix 
myF &lt;- read.table(paste(&#39;../data/Kinship_matrices/F_&#39;,mytissue,&#39;.txt&#39;,sep=&quot;&quot;))
## Get Eigen Values and Vectors 
myE &lt;- eigen(myF)
E_vectors &lt;- myE$vectors
E_values &lt;- myE$values
## Testing for selection on first 5 PCs
myM = 1:5
## Using the last 1/2 of PCs to estimate Va
myL = 6:dim(myF)[1]
### test for selection on all modules
moduleSelection &lt;- apply(moduleExpression, 2, function(x){
meanCenteredX = x[-length(x)] - mean(x)
myQpc = calcQpc(myZ = meanCenteredX, myU = E_vectors, myLambdas = E_values, myL = myL, myM = myM)
return(myQpc$pvals)
})
##make a dataframe with info to return
mydf = data.frame(t(moduleSelection))
names(mydf) = c(&#39;PC1&#39;,&#39;PC2&#39;,&#39;PC3&#39;,&#39;PC4&#39;,&#39;PC5&#39;)
mydf$module = row.names(mydf)
row.names(mydf) &lt;- NULL
mydflong = tidyr::gather(mydf, &#39;PC&#39;,&#39;pval&#39;, PC1:PC5)
mydflong$tissue = mytissue
return(mydflong)})

return(alltissuemodules)
}</code></pre>
<p>We have the p-values, let’s calculate the FDR and look for significant p-values.</p>
<pre class="r"><code>alltissuemodules = cluster_analysis_func(modules)

#combine all into one list
bigdf &lt;- do.call(&#39;rbind&#39;, alltissuemodules)

bigdf$fdr &lt;- p.adjust(bigdf$pval, method=&#39;fdr&#39;) ##calculate an FDR
summary(bigdf$fdr) ##nothing is significant</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.4299  0.9996  0.9996  0.9990  0.9996  0.9996 </code></pre>
<pre class="r"><code>kable(bigdf[which.min(bigdf[,3]),])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">module</th>
<th align="left">PC</th>
<th align="right">pval</th>
<th align="left">tissue</th>
<th align="right">fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">732</td>
<td align="left">RNA_Root_Meristem</td>
<td align="left">PC5</td>
<td align="right">0.0002409</td>
<td align="left">LMAD26</td>
<td align="right">0.4299294</td>
</tr>
</tbody>
</table>
<p>There is no significant selection on coexpression clusters. The lowest FDR is for the “RNA Root Meristem” cluster in adult leaf tissue on PC 5. We can also look at the clusters with with lowest bonferroni corrected p-values</p>
<pre class="r"><code>t &lt;- bigdf %&gt;% group_by(PC, tissue) %&gt;% mutate(bonferroni = p.adjust(pval, method=&#39;bonferroni&#39;)) %&gt;% arrange(bonferroni)
kable(head(t))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">module</th>
<th align="left">PC</th>
<th align="right">pval</th>
<th align="left">tissue</th>
<th align="right">fdr</th>
<th align="right">bonferroni</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">RNA_Root_Meristem</td>
<td align="left">PC5</td>
<td align="right">0.0002409</td>
<td align="left">LMAD26</td>
<td align="right">0.4299294</td>
<td align="right">0.0122837</td>
</tr>
<tr class="even">
<td align="left">RNA_FemaleSpikIt_._VegMeristem</td>
<td align="left">PC1</td>
<td align="right">0.0012924</td>
<td align="left">Kern</td>
<td align="right">0.7985099</td>
<td align="right">0.0659101</td>
</tr>
<tr class="odd">
<td align="left">Protein_Germinating_Kernals</td>
<td align="left">PC1</td>
<td align="right">0.0017164</td>
<td align="left">Kern</td>
<td align="right">0.7985099</td>
<td align="right">0.0875356</td>
</tr>
<tr class="even">
<td align="left">RNA_Root_Cortex_2</td>
<td align="left">PC1</td>
<td align="right">0.0017894</td>
<td align="left">Kern</td>
<td align="right">0.7985099</td>
<td align="right">0.0912583</td>
</tr>
<tr class="odd">
<td align="left">Protein_EndoCrown.PericarpAl</td>
<td align="left">PC1</td>
<td align="right">0.0039731</td>
<td align="left">Kern</td>
<td align="right">0.9996316</td>
<td align="right">0.2026281</td>
</tr>
<tr class="even">
<td align="left">RNA_Root_Cortex</td>
<td align="left">PC1</td>
<td align="right">0.0053132</td>
<td align="left">Kern</td>
<td align="right">0.9996316</td>
<td align="right">0.2709719</td>
</tr>
</tbody>
</table>
</div>
<div id="clusters-from-schaefer-et-al." class="section level2">
<h2>Clusters from Schaefer et al.</h2>
<p>ZmRoot Clusters - get clustes with &gt;= 100 genes</p>
<pre class="r"><code>clusters &lt;- fread(&quot;../data/schaefer_clusters.csv&quot;)
clusters &lt;- clusters[,1:4]

dat &lt;- clusters %&gt;% group_by(ZmRoot) %&gt;% mutate(num_genes = n()) %&gt;% filter(num_genes &gt;= 100)
names &lt;- unique(dat$ZmRoot)

gene_list &lt;- list()
max &lt;- nrow(dat[dat$ZmRoot == 1,])
for(i in 1:length(names)) {
  df &lt;- clusters %&gt;% filter(ZmRoot == names[i])
  genes &lt;- c(df$V1, rep(NA, max-nrow(df)))
  gene_list[[i]] &lt;- genes
}

modules &lt;- do.call(cbind, gene_list)</code></pre>
<p>Run analysis and FDR correction</p>
<pre class="r"><code>alltissuemodules = cluster_analysis_func(modules)

#combine all into one list
bigdf &lt;- do.call(&#39;rbind&#39;, alltissuemodules)

bigdf$fdr &lt;- p.adjust(bigdf$pval, method=&#39;fdr&#39;) ##calculate an FDR
summary(bigdf$fdr) ##nothing is significant</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.4377  0.9951  0.9951  0.9889  0.9951  0.9951 </code></pre>
<pre class="r"><code>kable(bigdf[which.min(bigdf[,3]),])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">module</th>
<th align="left">PC</th>
<th align="right">pval</th>
<th align="left">tissue</th>
<th align="right">fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">158</td>
<td align="left">4</td>
<td align="left">PC5</td>
<td align="right">0.0011368</td>
<td align="left">LMAD26</td>
<td align="right">0.4376829</td>
</tr>
</tbody>
</table>
<p>Look at lowest p-value clusters</p>
<pre class="r"><code>t &lt;- bigdf %&gt;% group_by(PC, tissue) %&gt;% mutate(bonferroni = p.adjust(pval, method=&#39;bonferroni&#39;)) %&gt;% arrange(bonferroni)
kable(head(t))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">module</th>
<th align="left">PC</th>
<th align="right">pval</th>
<th align="left">tissue</th>
<th align="right">fdr</th>
<th align="right">bonferroni</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">4</td>
<td align="left">PC5</td>
<td align="right">0.0011368</td>
<td align="left">LMAD26</td>
<td align="right">0.4376829</td>
<td align="right">0.0125052</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">PC5</td>
<td align="right">0.0023537</td>
<td align="left">LMAD26</td>
<td align="right">0.4530808</td>
<td align="right">0.0258903</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">PC1</td>
<td align="right">0.0053389</td>
<td align="left">Kern</td>
<td align="right">0.5741055</td>
<td align="right">0.0587274</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="left">PC2</td>
<td align="right">0.0063947</td>
<td align="left">LMAD26</td>
<td align="right">0.5741055</td>
<td align="right">0.0703414</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">PC3</td>
<td align="right">0.0074559</td>
<td align="left">LMAN26</td>
<td align="right">0.5741055</td>
<td align="right">0.0820151</td>
</tr>
<tr class="even">
<td align="left">11</td>
<td align="left">PC1</td>
<td align="right">0.0186208</td>
<td align="left">Kern</td>
<td align="right">0.9950821</td>
<td align="right">0.2048286</td>
</tr>
</tbody>
</table>
<p>ZmSAM Clusters - get clustes with &gt;= 100 genes</p>
<pre class="r"><code>dat &lt;- clusters %&gt;% group_by(ZmSAM) %&gt;% mutate(num_genes = n()) %&gt;% filter(num_genes &gt;= 100)
names &lt;- unique(dat$ZmSAM)

gene_list &lt;- list()
max &lt;- nrow(dat[dat$ZmSAM == 0,])
for(i in 1:length(names)) {
  df &lt;- clusters %&gt;% filter(ZmSAM == names[i])
  genes &lt;- c(df$V1, rep(NA, max-nrow(df)))
  gene_list[[i]] &lt;- genes
}

modules &lt;- do.call(cbind, gene_list)</code></pre>
<p>Run analysis and FDR correction</p>
<pre class="r"><code>alltissuemodules = cluster_analysis_func(modules)

#combine all into one list
bigdf &lt;- do.call(&#39;rbind&#39;, alltissuemodules)

bigdf$fdr &lt;- p.adjust(bigdf$pval, method=&#39;fdr&#39;) ##calculate an FDR
summary(bigdf$fdr) ##nothing is significant</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.9999  0.9999  0.9999  0.9999  0.9999  0.9999 </code></pre>
<pre class="r"><code>kable(bigdf[which.min(bigdf[,3]),])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">module</th>
<th align="left">PC</th>
<th align="right">pval</th>
<th align="left">tissue</th>
<th align="right">fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">184</td>
<td align="left">2</td>
<td align="left">PC5</td>
<td align="right">0.0033344</td>
<td align="left">LMAD26</td>
<td align="right">0.9999109</td>
</tr>
</tbody>
</table>
<p>Look at lowest p-value clusters</p>
<pre class="r"><code>t &lt;- bigdf %&gt;% group_by(PC, tissue) %&gt;% mutate(bonferroni = p.adjust(pval, method=&#39;bonferroni&#39;)) %&gt;% arrange(bonferroni)
kable(head(t))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">module</th>
<th align="left">PC</th>
<th align="right">pval</th>
<th align="left">tissue</th>
<th align="right">fdr</th>
<th align="right">bonferroni</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">2</td>
<td align="left">PC5</td>
<td align="right">0.0033344</td>
<td align="left">LMAD26</td>
<td align="right">0.9999109</td>
<td align="right">0.0433477</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="left">PC3</td>
<td align="right">0.0315067</td>
<td align="left">LMAN26</td>
<td align="right">0.9999109</td>
<td align="right">0.4095866</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">PC1</td>
<td align="right">0.0320505</td>
<td align="left">Kern</td>
<td align="right">0.9999109</td>
<td align="right">0.4166559</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">PC5</td>
<td align="right">0.0436062</td>
<td align="left">LMAD26</td>
<td align="right">0.9999109</td>
<td align="right">0.5668812</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">PC1</td>
<td align="right">0.0536048</td>
<td align="left">Kern</td>
<td align="right">0.9999109</td>
<td align="right">0.6968619</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">PC5</td>
<td align="right">0.0560459</td>
<td align="left">LMAN26</td>
<td align="right">0.9999109</td>
<td align="right">0.7285967</td>
</tr>
</tbody>
</table>
<p>ZmPAN Clusters - get clustes with &gt;= 100 genes</p>
<pre class="r"><code>dat &lt;- clusters %&gt;% group_by(ZmPAN) %&gt;% mutate(num_genes = n()) %&gt;% filter(num_genes &gt;= 100)
names &lt;- unique(dat$ZmPAN)

gene_list &lt;- list()
max &lt;- nrow(dat[dat$ZmPAN == 0,])
for(i in 1:length(names)) {
  df &lt;- clusters %&gt;% filter(ZmPAN == names[i])
  genes &lt;- c(df$V1, rep(NA, max-nrow(df)))
  gene_list[[i]] &lt;- genes
}

modules &lt;- do.call(cbind, gene_list)</code></pre>
<p>Run analysis and FDR correction</p>
<pre class="r"><code>alltissuemodules = cluster_analysis_func(modules)

#combine all into one list
bigdf &lt;- do.call(&#39;rbind&#39;, alltissuemodules)

bigdf$fdr &lt;- p.adjust(bigdf$pval, method=&#39;fdr&#39;) ##calculate an FDR
summary(bigdf$fdr) ##nothing is significant</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.9992  0.9992  0.9992  0.9992  0.9992  0.9992 </code></pre>
<pre class="r"><code>kable(bigdf[which.min(bigdf[,3]),])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">module</th>
<th align="left">PC</th>
<th align="right">pval</th>
<th align="left">tissue</th>
<th align="right">fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">241</td>
<td align="left">3</td>
<td align="left">PC5</td>
<td align="right">0.0045783</td>
<td align="left">LMAD26</td>
<td align="right">0.9991676</td>
</tr>
</tbody>
</table>
<p>Look at lowest p-value clusters</p>
<pre class="r"><code>t &lt;- bigdf %&gt;% group_by(PC, tissue) %&gt;% mutate(bonferroni = p.adjust(pval, method=&#39;bonferroni&#39;)) %&gt;% arrange(bonferroni)
kable(head(t))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">module</th>
<th align="left">PC</th>
<th align="right">pval</th>
<th align="left">tissue</th>
<th align="right">fdr</th>
<th align="right">bonferroni</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">3</td>
<td align="left">PC5</td>
<td align="right">0.0045783</td>
<td align="left">LMAD26</td>
<td align="right">0.9991676</td>
<td align="right">0.0778318</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">PC5</td>
<td align="right">0.0090431</td>
<td align="left">GShoot</td>
<td align="right">0.9991676</td>
<td align="right">0.1537334</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">PC5</td>
<td align="right">0.0092036</td>
<td align="left">LMAN26</td>
<td align="right">0.9991676</td>
<td align="right">0.1564620</td>
</tr>
<tr class="even">
<td align="left">13</td>
<td align="left">PC1</td>
<td align="right">0.0129806</td>
<td align="left">Kern</td>
<td align="right">0.9991676</td>
<td align="right">0.2206705</td>
</tr>
<tr class="odd">
<td align="left">13</td>
<td align="left">PC5</td>
<td align="right">0.0142503</td>
<td align="left">LMAD26</td>
<td align="right">0.9991676</td>
<td align="right">0.2422548</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">PC2</td>
<td align="right">0.0157494</td>
<td align="left">LMAN26</td>
<td align="right">0.9991676</td>
<td align="right">0.2677405</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.29        data.table_1.12.8 dplyr_1.0.0       quaint_0.0.0.9000
[5] ggpubr_0.3.0      ggplot2_3.3.2     reshape2_1.4.4    workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0 xfun_0.15        purrr_0.3.4      haven_2.3.1     
 [5] lattice_0.20-41  carData_3.0-4    colorspace_1.4-1 vctrs_0.3.1     
 [9] generics_0.0.2   htmltools_0.5.0  yaml_2.2.1       rlang_0.4.6     
[13] later_1.1.0.1    pillar_1.4.4     foreign_0.8-72   glue_1.4.1      
[17] withr_2.2.0      readxl_1.3.1     lifecycle_0.2.0  plyr_1.8.6      
[21] stringr_1.4.0    cellranger_1.1.0 munsell_0.5.0    ggsignif_0.6.0  
[25] gtable_0.3.0     zip_2.0.4        evaluate_0.14    rio_0.5.16      
[29] forcats_0.5.0    httpuv_1.5.4     curl_4.3         highr_0.8       
[33] broom_0.5.6      Rcpp_1.0.4.6     promises_1.1.1   scales_1.1.1    
[37] backports_1.1.8  abind_1.4-5      fs_1.4.1         hms_0.5.3       
[41] digest_0.6.25    stringi_1.4.6    openxlsx_4.1.5   rstatix_0.6.0   
[45] grid_3.6.2       rprojroot_1.3-2  tools_3.6.2      magrittr_1.5    
[49] tibble_3.0.1     crayon_1.3.4     whisker_0.4      tidyr_1.1.0     
[53] car_3.0-8        pkgconfig_2.0.3  ellipsis_0.3.1   rmarkdown_2.3   
[57] R6_2.4.1         nlme_3.1-148     git2r_0.27.1     compiler_3.6.2  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
