<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jennifer Blanc" />


<title>Selection_on_expression_of_coexpression_clusters</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Blancetal</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jgblanc/Blancetal">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Selection_on_expression_of_coexpression_clusters</h1>
<h4 class="author">Jennifer Blanc</h4>
<h4 class="date">3/13/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-03-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Blancetal/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200217code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200217)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200217code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200217)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjgblancBlancetaltreedf4fe40063954a4423f40f617a94fbbe5cb3842dtargetblankdf4fe40a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jgblanc/Blancetal/tree/df4fe40063954a4423f40f617a94fbbe5cb3842d" target="_blank">df4fe40</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjgblancBlancetaltreedf4fe40063954a4423f40f617a94fbbe5cb3842dtargetblankdf4fe40a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/Identifying_Selected_Genes/.DS_Store
    Ignored:    output/Selection_on_Expression_of_Cold_Response_Genes/.DS_Store
    Ignored:    output/Selection_on_expression_of_coexpression_clusters/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/df4fe40063954a4423f40f617a94fbbe5cb3842d/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">df4fe40</a>
</td>
<td>
jgblanc
</td>
<td>
2020-03-16
</td>
<td>
added more pvals
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgblanc/Blancetal/blob/2cb89cd06d2ebc12c3eb3103c6b0c5b78c690ba8/analysis/Selection_on_expression_of_coexpreession_clusters.Rmd" target="_blank">2cb89cd</a>
</td>
<td>
jgblanc
</td>
<td>
2020-03-13
</td>
<td>
adding coexpression stuff
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Here is the code to reproduce figure 4 and figure 5.</p>
</div>
<div id="code" class="section level2">
<h2>Code</h2>
<p>For this analysis we want to test for selection within specific coexpression modules. We used coexpression modules from Walley et al. (2016) who used weight gene coexpression network analysis (WGCNA) to to group together genes that were similarly expressed in at least 4 tissues in one maize inbred line. Their analysis resulted in 66 co-expression networks. Below we will load their co-expression networks and select all the clusters that have at least 100 genes in them.</p>
<pre class="r"><code>modules &lt;- read.delim(&quot;../data/Modules.txt&quot;,na.strings=c(&quot;&quot;,&quot;NA&quot;))
num_genes &lt;- apply(modules, 2, function(x) length(which(!is.na(x))))
num_genes_100 &lt;- which(num_genes &gt;= 100)
modules  &lt;- modules[,num_genes_100]</code></pre>
<p>We now have 51 modules that have at least 100 genes. We now need to the the names of the genes that we have in out RNA-seq data set. To do this we will read in the header for each of the mean centered expression matrices.</p>
<pre class="r"><code>kern_inds &lt;- t(read.table(&quot;../data/Mean_centered_expression/Kern.txt&quot;, nrows = 1))
gshoot_inds &lt;- t(read.table(&quot;../data/Mean_centered_expression/GShoot.txt&quot;, nrows = 1))
groot_inds &lt;- t(read.table(&quot;../data/Mean_centered_expression/GRoot.txt&quot;, nrows = 1))
base_inds &lt;- t(read.table(&quot;../data/Mean_centered_expression/L3Base.txt&quot;, nrows = 1))
tip_inds &lt;- t(read.table(&quot;../data/Mean_centered_expression/L3Tip.txt&quot;, nrows = 1))
lmad8_inds &lt;- t(read.table(&quot;../data/Mean_centered_expression/LMAD8.txt&quot;, nrows = 1))
lman8_inds &lt;- t(read.table(&quot;../data/Mean_centered_expression/LMAN8.txt&quot;, nrows = 1))
lmad26_inds &lt;- t(read.table(&quot;../data/Mean_centered_expression/LMAD26.txt&quot;, nrows = 1))
lman26_inds &lt;- t(read.table(&quot;../data/Mean_centered_expression/LMAN26.txt&quot;, nrows = 1))</code></pre>
<p>Now we have the list of gene names in each of the 51 coexpression modules and the name of the genes we have RNA seq data from in each of our nine tissue types. Now we need to find the overlap between our data and the genes in the sub modules. The chuck of code below can be used to get the overlap for any tissue, you just need to change “kern_inds” to be the tissue from above that you are interested in.</p>
<pre class="r"><code>tissue_list &lt;- list()
tissue_list_length &lt;- list()
for (i in 1:ncol(modules)) {
  lis &lt;- modules[,i]
  olap &lt;- Reduce(intersect, list(kern_inds[,1], lis)) # Change tissue 
  tissue_list[[i]] &lt;- olap
  tissue_list_length[[i]] &lt;- length(olap)
}</code></pre>
<p>Now the we have the overlap of gene name between our data and the coexpression modules, we now need find the median expression of all the overlap genes in each coexpression cluster. We also only want to keep clusters where the overlap number of genes is at least 50. The code below can be modified for all the tissues, the resulting tables for all nine tissues with the median expression of each coexpression clusters are save in “output/Selection_on_expression_of_coexpression_clusters/”.</p>
<pre class="r"><code>exp &lt;- read.table(&quot;../data/Raw_expression/Kern.txt&quot;) # Change tissue
median_exp &lt;- as.data.frame(matrix(ncol=51, nrow=nrow(exp)))
for (i in 1:ncol(modules)) {
  gene &lt;- tissue_list[[i]] # Make sure you ran correct tissue above 
  avg_exp &lt;- as.data.frame(matrix(nrow=nrow(exp), ncol = length(gene)))
    for (k in 1:length(gene)) {
      nam &lt;- gene[k]
      avg_exp[,k] &lt;- as.numeric(exp[,nam])}
    for (j in 1:nrow(exp)){
      if (ncol(avg_exp) &gt;= 50){
        e &lt;- t(avg_exp[j,])
        median_exp[j,i] &lt;- as.numeric(median(e[,1]))}
      else {
        median_exp[j,i] &lt;- NA}
    }
}

#write.table(median_exp,&quot;../output/Selection_on_expression_of_coexpression_clusters/Median_Kern.txt&quot;)</code></pre>
<p>Great, now we have the median expression value for each cluster. We now want to test for selection on the median values in each cluster. Below is the same code used to identify selected genes, we are simply conducting the test on median expression values in each line instead of a single gene expression in each line. For the test we are combining PC’s 1-5 for the numerator of the selection test first (saved as output/Selection_on_expression_of_coexpression_clusters/tissuename_pvals.txt) and each PC individually next (saved as output/Selection_on_expression_of_coexpression_clusters/tissuename_pvals_ind.txt)</p>
<pre class="r"><code>#df1 &lt;- median_exp
df1 &lt;- read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/Median_Kern.txt&quot;) #Change tissue 

## Mean center the data 
for (i in 1:ncol(df1)){
  df1[,i] &lt;- scale(df1[,i], scale = FALSE)
}

## Remove last row
df1 &lt;- df1[-nrow(df1),]

## Kinship Matrix for all 207 Kern lines - Change for different tissue
myF &lt;- read.table(&#39;../data/Kinship_matrices/F_Kern.txt&#39;)

## Get Eigen Values and Vectors 
myE &lt;- eigen(myF)
E_vectors &lt;- myE$vectors
E_values &lt;- myE$values

## Make new matrix to collect Z values
df2 &lt;- data.frame(matrix(ncol=ncol(df1), nrow=nrow(df1)))
colnames(df2) &lt;- colnames(df1[1:ncol(df1)])
rownames(df2) &lt;- rownames(df1)

## Calculate Q values by multiplying the mean-centered expression value by each eigen vector 
for (i in 1:ncol(df2)) {
  mean_centered_data &lt;- t(as.matrix(as.numeric(df1[,i])))
  for (k in 1:nrow(df2)){
    u &lt;- as.matrix(as.numeric(E_vectors[,k]))
    value &lt;- mean_centered_data %*% u
    df2[k,i] &lt;- value
  }
}

## Get the square root of the Eigen values   
de &lt;- data.frame(matrix(nrow = nrow(df1),ncol = 2))
de$Egien_values &lt;- E_values 
de$Sqrt_EV &lt;- sqrt(de$Egien_values)

## Calculate C-values by dividing Q values by the square root of the eigen values
df4 &lt;- data.frame(matrix(ncol=ncol(df2),nrow=nrow(df2)))
for (i in 1:ncol(df2)){
  df4[,i] &lt;- (df2[,i] / de$Sqrt_EV)
}

## Calculate F-values by dividing variances - for PCs 1-5 combined
F_values &lt;- data.frame(matrix(ncol=ncol(df2), nrow = 1))
for (j in 1:ncol(df2)){
  for (i in 1:1){
    q &lt;- df4[1:5,j] 
    t &lt;- df4[11:20,j]
    var_q &lt;- mean(q^2)
    var_t &lt;- mean(t^2)
    F_value &lt;- var_q / var_t
    F_values[i,j] &lt;- F_value
  }
}

## Calculate P-values from recorded F values 
P_values &lt;- data.frame(matrix(ncol=ncol(df2), nrow =1))
for (j in 1:ncol(F_values)){
  for (r in 1:1) {
    f_stat &lt;- F_values[r, j]
    p_value &lt;- pf(q=f_stat, df1=5, df2=10, lower.tail=FALSE) 
    P_values[r, j] &lt;- p_value
  }
}

#write.table(P_values,&quot;../output/Selection_on_expression_of_coexpression_clusters/Kern_pvals.txt&quot;)

## Calculate F-values by dividing variances 
F_values &lt;- data.frame(matrix(ncol=ncol(df2), nrow = 5))
for (j in 1:ncol(df2)){
  for (i in 1:5){
    q &lt;- df4[i,j] 
    t &lt;- df4[11:20,j]
    var_q &lt;- (q^2)
    var_t &lt;- mean(t^2)
    F_value &lt;- var_q / var_t
    F_values[i,j] &lt;- F_value
  }
}

## Calculate P-values from recorded F values 
P_values &lt;- data.frame(matrix(ncol=ncol(df2), nrow =1))
for (j in 1:ncol(F_values)){
  for (r in 1:5) {
    f_stat &lt;- F_values[r, j]
    p_value &lt;- pf(q=f_stat, df1=1, df2=10, lower.tail=FALSE) 
    P_values[r, j] &lt;- p_value
  }
}

#write.table(P_values,&quot;../output/Selection_on_expression_of_coexpression_clusters/Kern_pvals_ind.txt&quot;)</code></pre>
</div>
<div id="figures" class="section level2">
<h2>Figures</h2>
<p>We now have the raw p-values from the results of our selection test. Let’s plot the figure 4 from the main text using these p-values.</p>
<p>First let’s read in the p-values.</p>
<pre class="r"><code>kern &lt;- t(read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/Kern_pvals.txt&quot;))
gshoot &lt;- t(read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/GShoot_pvals.txt&quot;))
groot &lt;- t(read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/GRoot_pvals.txt&quot;))
base &lt;- t(read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/L3Base_pvals.txt&quot;))
tip &lt;- t(read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/L3Tip_pvals.txt&quot;))
lmad8 &lt;- t(read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/LMAD8_pvals.txt&quot;))
lman8 &lt;- t(read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/LMAN8_pvals.txt&quot;))
lmad26 &lt;- t(read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/LMAD26_pvals.txt&quot;))
lman26 &lt;- t(read.table(&quot;../output/Selection_on_expression_of_coexpression_clusters/LMAN26_pvals.txt&quot;))</code></pre>
<p>We need to add the module names as the row names.</p>
<pre class="r"><code>names &lt;- colnames(modules)
rownames(kern) &lt;- names
colnames(kern) &lt;- c(&quot;Kern&quot;)
rownames(groot) &lt;- names
colnames(groot) &lt;- c(&quot;GRoot&quot;)
rownames(gshoot) &lt;- names
colnames(gshoot) &lt;- c(&quot;GShoot&quot;)
rownames(base) &lt;- names
colnames(base) &lt;- c(&quot;L3Base&quot;)
rownames(tip) &lt;- names
colnames(tip) &lt;- c(&quot;L3Tip&quot;)
rownames(lmad8) &lt;- names
colnames(lmad8) &lt;- c(&quot;LMAD8&quot;)
rownames(lman8) &lt;- names
colnames(lman8) &lt;- c(&quot;LMAN8&quot;)
rownames(lmad26) &lt;- names
colnames(lmad26) &lt;- c(&quot;LMAD26&quot;)
rownames(lman26) &lt;- names
colnames(lman26) &lt;- c(&quot;LMAN26&quot;)</code></pre>
<p>Now we will check the q-value for each test using the FDR multiple testing correction.</p>
<pre class="r"><code>kern_q &lt;- p.adjust(kern, method = &quot;fdr&quot;)
gshoot_q &lt;- p.adjust(gshoot, method = &quot;fdr&quot;)
groot_q &lt;- p.adjust(groot, method = &quot;fdr&quot;)
base_q &lt;- p.adjust(base, method = &quot;fdr&quot;)
tip_q &lt;- p.adjust(tip, method = &quot;fdr&quot;)
lmad8_q &lt;- p.adjust(lmad8, method = &quot;fdr&quot;)
lman8_q &lt;- p.adjust(lman8, method = &quot;fdr&quot;)
lmad26_q &lt;- p.adjust(lmad26, method = &quot;fdr&quot;)
lman26_q &lt;- p.adjust(lman26, method = &quot;fdr&quot;)

dat_q &lt;- cbind(kern_q, groot_q, gshoot_q, base_q, tip_q, lmad8_q, lman8_q, lmad26_q, lman26_q)
dat_q &lt;- as.data.frame(dat_q)
dat_q$Module &lt;- names
dat_q &lt;- dat_q[-c(29, 15, 36), ] # Remove modules that did not have at least 50 genes in any of the tissues 
dat_q &lt;- melt(dat_q,id=&quot;Module&quot;)
colnames(dat_q) &lt;- c(&quot;Module&quot;, &quot;Tissue&quot;, &quot;QVAL&quot;)</code></pre>
<p>Now we will combine the p-values into one data frame and add a factor column that annotates the p-value range</p>
<pre class="r"><code>dat &lt;- cbind(kern, groot, gshoot, base, tip, lmad8, lman8, lmad26, lman26)
dat &lt;- as.data.frame(dat)
dat$Module &lt;- rownames(dat)

dat &lt;- dat[-c(29, 15,36), ] # Remove modules that did not have at least 50 genes in any of the tissues 

dat2 &lt;- melt(dat,id=&quot;Module&quot;)
dat2[is.na(dat2)] &lt;- 1
dat2$cat &lt;- &quot;(0.1,1]&quot;

for (i in 1:nrow(dat2)) {
  p &lt;- dat2[i,3]
  if (p  == 1) {dat2[i,4] &lt;- &quot;NA&quot; }
  else if (p &gt; 0.001 &amp; p &lt;= 0.01 ) { dat2[i,4] &lt;- &quot;(0,0.01]&quot;}
  else if (p &gt; 0.01 &amp; p &lt;= 0.05) {dat2[i,4] &lt;- &quot;(0.01,0.05]&quot;}
  else if (p &gt; 0.05 &amp; p &lt;= 0.1) {dat2[i,4] &lt;- &quot;(0.05,0.1]&quot;}
}</code></pre>
<p>Finally we will join the p-value and q-value information and plot the heat map. The asterisks are marking q-value ranges, one module has a q-value below 0.1.</p>
<pre class="r"><code>dat3 &lt;- cbind(dat2, dat_q)
dat3$stars &lt;- cut(dat3$QVAL, breaks=c(-Inf, 0.01, 0.05, 0.1, Inf), label=c(&quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;))
dat3 &lt;- dat3[, -5]

pl &lt;- ggplot(data=dat3, aes(x=Module,y=variable, fill = cat)) + geom_tile() + scale_fill_manual(values = c(&quot;#D55E00&quot;, &#39;#56B4E9&#39;, &quot;#F0E442&quot;, &quot;white&quot;, &quot;gray95&quot;), name = &quot;p-value range&quot;) + theme_bw() + xlab(&quot;Module&quot;) + ylab(&quot;Tissue&quot;) + theme(axis.text.x = element_text(angle = 45, hjust = 1), axis.text.y = element_text(size = 7),legend.key = element_rect(colour = &#39;black&#39;, size = 0.5), axis.title.y = element_text(size = 14), axis.title.x = element_text(size = 14)) + coord_flip() + geom_text(aes(label=stars), color=&quot;black&quot;, size=5, nudge_x = -.5)
pl</code></pre>
<p><img src="figure/Selection_on_expression_of_coexpreession_clusters.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Finally we will plot the expression plot Median RNA Root Cortex coexpression cluster.</p>
<p>Now we have all the info to make the expression plots. It is important to note that we calculated the <span class="math inline">\(\text{var}(C_{11:20})\)</span> for this specific module in the kernel tissue when we were doing the selection test above and manually added used it to plot these confidence intervals. The code below can generate the expression plot at best fit line for all tissues and modules but NOT the correct confidence intervals. To plot the correct confidence interval you would need to record the “var_t” value for each module from the code above and plug that into the code below used to calculate “k_plus”.</p>
<pre class="r"><code># PC 1
pop_dat$one &lt;- E_vectors[,1]
lambda &lt;- E_values[1]

k_plus &lt;-  1.96 * sqrt(9.137196 * lambda) # Excpected denominator calculated from script 
k_minus &lt;- - k_plus 

lR &lt;- lm(pop_dat$RNA_Root_Cortex ~ pop_dat$one) 
coeff &lt;- lR$coefficients[[2]]

col &lt;-c(&#39;#E69F00&#39;, &#39;#56B4E9&#39;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)
pl1 &lt;- ggplot(data=pop_dat, aes(x = one, y= RNA_Root_Cortex  , color=Subpopulation)) + scale_colour_manual(values = col, labels=c(&quot;mixed&quot;, &quot;non-stiff stalk&quot;, &quot;popcorn&quot;, &quot;stiff stalk&quot;, &quot;sweet&quot;, &quot;tropical&quot;)) + xlab(&quot;PC 1&quot;) +  ylab(&quot;Expression&quot;) + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.y = element_text(size=12), axis.title.x  = element_text(size=12), legend.position = &quot;right&quot;, legend.title = element_text(size = 12), legend.text = element_text(size = 10)) + geom_point(size = 1.5) + geom_abline(slope = k_plus, linetype = 2) + geom_abline(slope = coeff, size = 1.5)+  geom_abline(slope = k_minus, linetype = 2)

# PC 2 
pop_dat$two &lt;- E_vectors[,2]
lambda &lt;- E_values[2]

k_plus &lt;-  1.96 * sqrt(9.137196 * lambda) # Excpected denominator calculated from script 
k_minus &lt;- - k_plus 

lR &lt;- lm(pop_dat$RNA_Root_Cortex ~ pop_dat$two) 
coeff &lt;- lR$coefficients[[2]]

col &lt;-c(&#39;#E69F00&#39;, &#39;#56B4E9&#39;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)
pl2 &lt;- ggplot(data=pop_dat, aes(x = two, y= RNA_Root_Cortex  , color=Subpopulation)) + scale_colour_manual(values = col, labels=c(&quot;mixed&quot;, &quot;non-stiff stalk&quot;, &quot;popcorn&quot;, &quot;stiff stalk&quot;, &quot;sweet&quot;, &quot;tropical&quot;)) + xlab(&quot;PC 1&quot;) +  ylab(&quot;Expression&quot;) + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.y = element_text(size=12), axis.title.x  = element_text(size=12), legend.position = &quot;right&quot;, legend.title = element_text(size = 12), legend.text = element_text(size = 10)) + geom_point(size = 1.5) + geom_abline(slope = k_plus, linetype = 2) + geom_abline(slope = coeff, size = 1.5)+  geom_abline(slope = k_minus, linetype = 2)

# PC 3
pop_dat$three &lt;- E_vectors[,3]
lambda &lt;- E_values[3]

k_plus &lt;-  1.96 * sqrt(9.137196 * lambda) # Excpected denominator calculated from script 
k_minus &lt;- - k_plus 

lR &lt;- lm(pop_dat$RNA_Root_Cortex ~ pop_dat$three) 
coeff &lt;- lR$coefficients[[2]]

col &lt;-c(&#39;#E69F00&#39;, &#39;#56B4E9&#39;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)
pl3 &lt;- ggplot(data=pop_dat, aes(x = three, y= RNA_Root_Cortex  , color=Subpopulation)) + scale_colour_manual(values = col, labels=c(&quot;mixed&quot;, &quot;non-stiff stalk&quot;, &quot;popcorn&quot;, &quot;stiff stalk&quot;, &quot;sweet&quot;, &quot;tropical&quot;)) + xlab(&quot;PC 1&quot;) +  ylab(&quot;Expression&quot;) + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.y = element_text(size=12), axis.title.x  = element_text(size=12), legend.position = &quot;right&quot;, legend.title = element_text(size = 12), legend.text = element_text(size = 10)) + geom_point(size = 1.5) + geom_abline(slope = k_plus, linetype = 2) + geom_abline(slope = coeff, size = 1.5)+  geom_abline(slope = k_minus, linetype = 2)

# PC 4
pop_dat$four &lt;- E_vectors[,4]
lambda &lt;- E_values[4]

k_plus &lt;-  1.96 * sqrt(9.137196 * lambda) # Excpected denominator calculated from script 
k_minus &lt;- - k_plus 

lR &lt;- lm(pop_dat$RNA_Root_Cortex ~ pop_dat$four) 
coeff &lt;- lR$coefficients[[2]]

col &lt;-c(&#39;#E69F00&#39;, &#39;#56B4E9&#39;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)
pl4 &lt;- ggplot(data=pop_dat, aes(x = four, y= RNA_Root_Cortex  , color=Subpopulation)) + scale_colour_manual(values = col, labels=c(&quot;mixed&quot;, &quot;non-stiff stalk&quot;, &quot;popcorn&quot;, &quot;stiff stalk&quot;, &quot;sweet&quot;, &quot;tropical&quot;)) + xlab(&quot;PC 1&quot;) +  ylab(&quot;Expression&quot;) + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.y = element_text(size=12), axis.title.x  = element_text(size=12), legend.position = &quot;right&quot;, legend.title = element_text(size = 12), legend.text = element_text(size = 10)) + geom_point(size = 1.5) + geom_abline(slope = k_plus, linetype = 2) + geom_abline(slope = coeff, size = 1.5)+  geom_abline(slope = k_minus, linetype = 2)

# PC 5
pop_dat$five &lt;- E_vectors[,5]
lambda &lt;- E_values[5]

k_plus &lt;-  1.96 * sqrt(9.137196 * lambda) # Excpected denominator calculated from script 
k_minus &lt;- - k_plus 

lR &lt;- lm(pop_dat$RNA_Root_Cortex ~ pop_dat$five) 
coeff &lt;- lR$coefficients[[2]]

col &lt;-c(&#39;#E69F00&#39;, &#39;#56B4E9&#39;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;)
pl5 &lt;- ggplot(data=pop_dat, aes(x = five, y= RNA_Root_Cortex  , color=Subpopulation)) + scale_colour_manual(values = col, labels=c(&quot;mixed&quot;, &quot;non-stiff stalk&quot;, &quot;popcorn&quot;, &quot;stiff stalk&quot;, &quot;sweet&quot;, &quot;tropical&quot;)) + xlab(&quot;PC 1&quot;) +  ylab(&quot;Expression&quot;) + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.y = element_text(size=12), axis.title.x  = element_text(size=12), legend.position = &quot;right&quot;, legend.title = element_text(size = 12), legend.text = element_text(size = 10)) + geom_point(size = 1.5) + geom_abline(slope = k_plus, linetype = 2) + geom_abline(slope = coeff, size = 1.5)+  geom_abline(slope = k_minus, linetype = 2)</code></pre>
<p>Finally plot them all together! <img src="figure/Selection_on_expression_of_coexpreession_clusters.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggpubr_0.2.5    magrittr_1.5    ggplot2_3.2.1   reshape2_1.4.3 
[5] workflowr_1.6.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3       compiler_3.6.2   pillar_1.4.3     later_1.0.0     
 [5] git2r_0.26.1     plyr_1.8.5       tools_3.6.2      digest_0.6.25   
 [9] evaluate_0.14    lifecycle_0.1.0  tibble_2.1.3     gtable_0.3.0    
[13] pkgconfig_2.0.3  rlang_0.4.4      yaml_2.2.1       xfun_0.12       
[17] gridExtra_2.3    withr_2.1.2      stringr_1.4.0    dplyr_0.8.4     
[21] knitr_1.28       fs_1.3.1         cowplot_1.0.0    rprojroot_1.3-2 
[25] grid_3.6.2       tidyselect_1.0.0 glue_1.3.1       R6_2.4.1        
[29] rmarkdown_2.1    farver_2.0.3     purrr_0.3.3      whisker_0.4     
[33] backports_1.1.5  scales_1.1.0     promises_1.1.0   htmltools_0.4.0 
[37] assertthat_0.2.1 colorspace_1.4-1 ggsignif_0.6.0   httpuv_1.5.2    
[41] labeling_0.3     stringi_1.4.6    lazyeval_0.2.2   munsell_0.5.0   
[45] crayon_1.3.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
