---
title: "Selection on Expression of Cold Response Genes"
author: "Jennifer Blanc"
date: "2/17/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(dplyr)
library(knitr)
library(reshape2)
```

## Intro  

The goal of this analysis is to determine if genes that have been shown to be differentially expressed in response to cold treatment are enriched for genes whose expression is under selection. 


## Code 

First, read in the Avila et al. differential expression data and get the names of the genes used in their analysis. 
```{r}
cold <- read.csv("../data/Cold.csv", header = F, stringsAsFactors = F)
cold <-cold[,2:19] 
colnames(cold) <- c("V3_Gene_Name","V4_Gene_Name","Arabidopsis_Ortholog","Rice_Ortholog","Cold_1D_CG60_Means","Cold_1D_CG102_Means","Control_1D_CG60_Means","Control_1D_CG102_Means","FDR_1D_Genotype","FDR_1D_Treatment","FDR_\
1D_Interaction","Cold_4D_CG60_Means","Cold_4D_CG102_Means","Control_4D_CG60_Means","Control_4D_CG102_Means","FDR_4D_Genotype","FDR_4D_Treatment","FDR_4D_Interaction")

cold_names <- cold$V3_Gene_Name
```

We need to pick only the genes for which we have expression data for. To do this we will load the names for the genes we have for each tissue and take the overlap with the Avila data.  
```{r}

alltissues = c('GRoot',"Kern","LMAD26","LMAN26", "L3Tip","GShoot","L3Base")

load('../data/quaint-results.rda')

```

The next step is to determine if differentially expressed genes are more likely to be under selection in every PC/Tissue combination. To do this we will use a chi-squared test.  

```{r, warning=FALSE, message=FALSE}
coldsum <- cold %>% select(V3_Gene_Name, V4_Gene_Name, FDR_1D_Treatment, FDR_4D_Treatment) ## pull out important columns

coldsum$sig = apply(coldsum[,3:4], 1, min, na.rm=T)
sigGenes = dplyr::filter(coldsum, sig < 0.1)$V3_Gene_Name
length(sigGenes)
notsigGenes = dplyr::filter(coldsum, sig > 0.1)$V3_Gene_Name
length(notsigGenes)


##function for running the chisq test on a specific pc/tissue combination
run_chi_sq <- function(mytissue, sigGenes, notsigGenes, myPC) {
  quaint_results = alltissueresults[[mytissue]] ##pull out tissue specific genes
  quaint_results_pc = data.frame(mygene = unlist(quaint_results[1,]), pval = unlist(lapply(quaint_results[5,], function(x){x[myPC]})), stringsAsFactors = F)  ## just need a table for the specific PC's pvalue
  sigOver = dplyr::filter(quaint_results_pc, mygene %in% sigGenes)
  notsigOver = dplyr::filter(quaint_results_pc, mygene %in% notsigGenes)
  ##make the chisq table
  bothsig = nrow(dplyr::filter(sigOver, pval < 0.05))
  diffexponly = nrow(dplyr::filter(sigOver, pval >= 0.05))
  selonly = nrow(dplyr::filter(notsigOver, pval < 0.05))
  nothingsig = nrow(dplyr::filter(notsigOver, pval > 0.05))
  chitbl = matrix(c(bothsig, selonly, diffexponly, nothingsig), nrow=2)
  chitest = chisq.test(chitbl)
  return(chisq.test$p.value)
}

##run on all PCs and all tissues for the cold genes


```


These are the undjusted p-values for our chi-square test for independence between the two categorical variables (Genes under selection and DE cold genes). From the p-value alone we don't know what direction the enrichment is in (Are DE genes more or less likely to be under selection). Below is the results of the chi-square test for the 5 PC's in each of the 9 tissues (Table S3) - this table can be generated my running the code above for each tissue type.  

```{r}
chi_all <- read.table("../output/Selection_on_Expression_of_Cold_Response_Genes/Chi_sqr_pvals.txt") 
kable(chi_all)
```


## Figure  

Now we want to plot Figure 3 in the main text. First we will look at LMAD26/PC5 which has a p-value of 0.00103 as an example of PC/tissue combination where selected genes are enriched for DE cold genes. To run this code we need the p-values and the contigency table of selected genes vs DE genes. The key to generating these figures for every tissue/PC combination is to load the correct p-values and use the "run_chi_sq" function to extract the contingency table. 

```{r,warning=FALSE, message=FALSE}
dat <- read.table("../output/Selection_on_Expression_of_Cold_Response_Genes/LMAD26.txt") # Pick Tissuee

Cold <- cold %>%
  select(V3_Gene_Name, V4_Gene_Name, FDR_1D_Treatment, FDR_4D_Treatment)
names <- as.data.frame(olap_lmad26) # Pick Tissue 
colnames(names) <- "V3_Gene_Name"
Cold <- inner_join(names, Cold)
Cold <- sig_DE(Cold) # Get which genes are DE

# Get contingency table
out <- run_chi_sq(Cold, t(dat))
dt <- out[[1]][[5]] # Pick correct contingency table 
df <- as.data.frame(as.table(dt))
```

Plot Bar Chart   
```{r}
colnames(df) <- c("CR", "Sel", "Freq")
df1 <- subset(df, Sel == TRUE)
df1[1,3] <- df1[1,3] / sum(df1[1,3] + df[1,3])
df1[2,3] <- df1[2,3] / sum(df1[2,3] + df[2,3])

col <- c( "#0072B2")
pl1 <- ggplot(data = df1, aes(x = CR, y = Freq)) + geom_bar(stat="identity", fill = col)  +  ylab("proportion p < 0.05") + xlab("") + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.y = element_text(size=12), axis.title.x  = element_text(size=12), legend.position = "right", legend.title = element_text(size = 12), legend.text = element_text(size = 10)) + ggtitle("LMAD26 PC 5") + theme(plot.title = element_text(hjust = 0.5, size = 14))

pl1
```

Plot Historgram  
```{r}
tdat <- as.data.frame(t(dat))
colnames(tdat) <- c("PC1", "PC2", "PC3", "PC4", "PC5")
df <- cbind(Cold, tdat)

mysig = subset(df, Sig==TRUE)
sighist = hist(mysig$PC5, plot=FALSE)
mynot = subset(df, Sig==FALSE)
nothist = hist(mynot$PC5, plot=FALSE)
combinedHist = rbind(nothist$density, sighist$density)

x <- t(combinedHist)
colnames(x) <- c("Not cold-response", "Cold-response")
m <- melt(x)

pl2 <- ggplot(data = m, aes(x = Var1/10, y = value, fill = Var2)) + geom_bar(stat="identity",position="dodge") + scale_fill_manual(name = "", values = c("#CC79A7",   "#F0E442")) + theme_classic() + ylab("proportion") + xlab("") +  scale_x_continuous(breaks=seq(0, 1, 0.1)) + theme(legend.position = c(0.8, 0.9)) + ggtitle("LMAD26 PC5") + theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title.y = element_text(size=12), legend.text = element_text(size=8), legend.key.size = unit(0.8, "line"))

pl2
```


Now we will plot Figure 3 CD
```{r,warning=FALSE, message=FALSE}
dat <- read.table("../output/Selection_on_Expression_of_Cold_Response_Genes/LMAD8.txt") # Pick Tissue

Cold <- cold %>%
  select(V3_Gene_Name, V4_Gene_Name, FDR_1D_Treatment, FDR_4D_Treatment)
names <- as.data.frame(olap_lmad8) # Pick Tissue 
colnames(names) <- "V3_Gene_Name"
Cold <- inner_join(names, Cold)
Cold <- sig_DE(Cold) # Get which genes are DE

# Get contingency table
out <- run_chi_sq(Cold, t(dat))
dt <- out[[1]][[4]] # Pick correct contingency table out[[1]][[x]] -> x is the PC you want to look at 
df <- as.data.frame(as.table(dt))
```

Plot Bar Chart   
```{r}
colnames(df) <- c("CR", "Sel", "Freq")
df1 <- subset(df, Sel == TRUE)
df1[1,3] <- df1[1,3] / sum(df1[1,3] + df[1,3])
df1[2,3] <- df1[2,3] / sum(df1[2,3] + df[2,3])

col <- c( "#0072B2")
pl1 <- ggplot(data = df1, aes(x = CR, y = Freq)) + geom_bar(stat="identity", fill = col)  +  ylab("proportion p < 0.05") + xlab("") + theme_bw() + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.y = element_text(size=12), axis.title.x  = element_text(size=12), legend.position = "right", legend.title = element_text(size = 12), legend.text = element_text(size = 10)) + ggtitle("LMAD8 PC 4") + theme(plot.title = element_text(hjust = 0.5, size = 14))

pl1
```

```{r}
tdat <- as.data.frame(t(dat))
colnames(tdat) <- c("PC1", "PC2", "PC3", "PC4", "PC5")
df <- cbind(Cold, tdat)

mysig = subset(df, Sig==TRUE)
sighist = hist(mysig$PC5, plot=FALSE)
mynot = subset(df, Sig==FALSE)
nothist = hist(mynot$PC5, plot=FALSE) # Change to correct PC
combinedHist = rbind(nothist$density, sighist$density)

x <- t(combinedHist)
colnames(x) <- c("Not cold-response", "Cold-response")
m <- melt(x)

pl2 <- ggplot(data = m, aes(x = Var1/10, y = value, fill = Var2)) + geom_bar(stat="identity",position="dodge") + scale_fill_manual(name = "", values = c("#CC79A7",   "#F0E442")) + theme_classic() + ylab("proportion") + xlab("") +  scale_x_continuous(breaks=seq(0, 1, 0.1)) + theme(legend.position = c(0.8, 0.9)) + ggtitle("LMAD8 PC4") + theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title.y = element_text(size=12), legend.text = element_text(size=8), legend.key.size = unit(0.8, "line"))

pl2
```
