---
title: "Selection_on_expression_of_coexpression_clusters"
author: "Jennifer Blanc"
date: "3/13/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(reshape2)
library(ggplot2)
library(ggpubr)
library(quaint)
```


## Intro  

Here is the code to test for expression on coeexpression clusters from Walleey et al.     

## Code 

For this analysis we want to test for selection within specific coexpression modules. We used coexpression modules from Walley et al. (2016) who used weight gene coexpression network analysis (WGCNA) to to group together genes that were similarly expressed in at least 4 tissues in one maize inbred line. Their analysis resulted in 66 co-expression networks. Below we will load their co-expression networks and select all the clusters that have at least 100 genes in them.  

```{r}
modules <- read.delim("../data/Modules.txt",na.strings=c("","NA"), stringsAsFactors = F)
num_genes <- apply(modules, 2, function(x) length(which(!is.na(x))))
num_genes_100 <- which(num_genes >= 100)
modules  <- modules[,num_genes_100]
```

We now have 51 modules that have at least 100 genes. We will now get the median expression of all the genes that are present in our RNAseq dataset for each module and then test for selection on the median values in each cluster. The code below is similar code used to identify selected genes, we are simply conducting the test on median expression values in each line instead of a single gene expression in each line. Again we test for selection on the first 5 PCs and use the last half of PCs to estimate $V_a$. Th function will return the p-values from the $Q_{pc}$ test for all tissues.      

```{r}
alltissues = c('GRoot',"Kern","LMAD26","LMAN26", "L3Tip","GShoot","L3Base")

alltissuemodules = lapply(alltissues, function(mytissue){
  
##get the names of genes we have expression data for in each tissue
exp <- read.table(paste("../data/Raw_expression/",mytissue,".txt", sep="")) #  read in expression level
geneNames = names(exp)

#get the median expression level for each module in this tissue
moduleExpression = apply(modules, 2, function(x){
  olap = x[x %in% geneNames] #get genes in the module tha we have expression data for
  moduleExp = dplyr::select(exp, all_of(olap)) #pull out the expression data for these genes
  medianExp = apply(moduleExp, 1, median)
  return(medianExp)
})

####identify selection in each of these modules
# Read in tissue specific kinship matrix 
myF <- read.table(paste('../data/Kinship_matrices/F_',mytissue,'.txt',sep=""))
## Get Eigen Values and Vectors 
myE <- eigen(myF)
E_vectors <- myE$vectors
E_values <- myE$values
## Testing for selection on first 5 PCs
myM = 1:5
## Using the last 1/2 of PCs to estimate Va
myL = 6:dim(myF)[1]
### test for selection on all modules
moduleSelection <- apply(moduleExpression, 2, function(x){
meanCenteredX = x[-length(x)] - mean(x)
myQpc = calcQpc(myZ = meanCenteredX, myU = E_vectors, myLambdas = E_values, myL = myL, myM = myM)
return(myQpc$pvals)
})
##make a dataframe with info to return
mydf = data.frame(t(moduleSelection))
names(mydf) = c('PC1','PC2','PC3','PC4','PC5')
mydf$module = row.names(mydf)
row.names(mydf) <- NULL
mydflong = tidyr::gather(mydf, 'PC','pval', PC1:PC5)
mydflong$tissue = mytissue
return(mydflong)})

```
 
We have the p-values, let's calculate the FDR and look for significant p-values. 
```{r}
#combine all into one list
bigdf <- do.call('rbind', alltissuemodules)

bigdf$fdr <- p.adjust(bigdf$pval, method='fdr') ##calculate an FDR
summary(bigdf$fdr) ##nothing is significant

bigdf[which.min(bigdf[,3]),]
```

There is no significant selection on coexpression clusters. The lowest FDR is for the "RNA Root Meristem" cluster in adult leaf tissue on PC 5.   
