---
title: "Selection on Expression of Cold Response Genes"
author: "Jennifer Blanc"
date: "2/17/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(dplyr)
library(knitr)
library(reshape2)
```

## Intro  

The goal of this analysis is to determine if genes that have been shown to be differentially expressed in response to cold treatment are enriched for genes whose expression is under selection. 


## Code 

First, read in the Avila et al. differential expression data and get the names of the genes used in their analysis. 
```{r}
cold <- read.csv("../data/Cold.csv", header = F, stringsAsFactors = F)
cold <-cold[,2:19] 
colnames(cold) <- c("V3_Gene_Name","V4_Gene_Name","Arabidopsis_Ortholog","Rice_Ortholog","Cold_1D_CG60_Means","Cold_1D_CG102_Means","Control_1D_CG60_Means","Control_1D_CG102_Means","FDR_1D_Genotype","FDR_1D_Treatment","FDR_\
1D_Interaction","Cold_4D_CG60_Means","Cold_4D_CG102_Means","Control_4D_CG60_Means","Control_4D_CG102_Means","FDR_4D_Genotype","FDR_4D_Treatment","FDR_4D_Interaction")

cold_names <- cold$V3_Gene_Name
```

We need to pick only the genes for which we have expression data for. To do this we will load the names for the genes we have for each tissue and take the overlap with the Avila data.  
```{r}

alltissues = c('GRoot',"Kern","LMAD26","LMAN26", "L3Tip","GShoot","L3Base")

load('../data/quaint-results.rda')

```

The next step is to determine if differentially expressed genes are more likely to be under selection in every PC/Tissue combination. To do this we will use a chi-squared test.  

```{r, warning=FALSE, message=FALSE}
coldsum <- cold %>% select(V3_Gene_Name, V4_Gene_Name, FDR_1D_Treatment, FDR_4D_Treatment) ## pull out important columns

coldsum$sig = apply(coldsum[,3:4], 1, min, na.rm=T)
coldsigGenes = dplyr::filter(coldsum, sig < 0.1)$V3_Gene_Name
length(coldsigGenes)
coldnotsigGenes = dplyr::filter(coldsum, sig > 0.1)$V3_Gene_Name
length(coldnotsigGenes)


##function for running the chisq test on a specific pc/tissue combination
run_chi_sq <- function(mytissue, myPC,sigGenes, notsigGenes, myp = 0.05) {
  quaint_results = alltissueresults[[mytissue]] ##pull out tissue specific genes
  quaint_results_pc = data.frame(mygene = unlist(quaint_results[1,]), pval = unlist(lapply(quaint_results[5,], function(x){x[myPC]})), stringsAsFactors = F)  ## just need a table for the specific PC's pvalue
  sigOver = dplyr::filter(quaint_results_pc, mygene %in% sigGenes)
  notsigOver = dplyr::filter(quaint_results_pc, mygene %in% notsigGenes)
  ##make the chisq table
  bothsig = nrow(dplyr::filter(sigOver, pval < myp))
  diffexponly = nrow(dplyr::filter(sigOver, pval >= myp))
  selonly = nrow(dplyr::filter(notsigOver, pval < myp))
  nothingsig = nrow(dplyr::filter(notsigOver, pval > myp))
  chitbl = matrix(c(bothsig, selonly, diffexponly, nothingsig), nrow=2) ##make the chisq table
  chitest = chisq.test(chitbl) ##run the test
  return(chitest$p.value)
}

##run on all PCs and all tissues for the cold genes
myresults = sapply(alltissues, function(mytissue){
  sapply(1:5,function(myPC){
    run_chi_sq(mytissue, myPC, coldsigGenes, coldnotsigGenes, 0.05)
  })
})

##do we really just care about combos with significant evidence of selection in at least one gene?

### note that we don't know what direction this goes in.



```

## Figure  

Now we want to plot Figure 3 in the main text. First we will look at LMAD26/PC5  as an example of PC/tissue combination where selected genes are enriched for DE cold genes. To run this code we need the p-values and the contigency table of selected genes vs DE genes. The key to generating these figures for every tissue/PC combination is to load the correct p-values and use the "run_chi_sq" function to extract the contingency table. 


Plot Bar Chart   
```{r}
myp = 0.05
quaint_results = alltissueresults[['LMAD26']] ##pull out tissue specific genes
quaint_results_pc = data.frame(mygene = unlist(quaint_results[1,]), pval = unlist(lapply(quaint_results[5,], function(x){x[5]})), stringsAsFactors = F)  ## just need a table for the specific PC's pvalue
sigOver = dplyr::filter(quaint_results_pc, mygene %in% sigGenes)
notsigOver = dplyr::filter(quaint_results_pc, mygene %in% notsigGenes)
##make the chisq table
bothsig = nrow(dplyr::filter(sigOver, pval < myp))
diffexponly = nrow(dplyr::filter(sigOver, pval >= myp))
selonly = nrow(dplyr::filter(notsigOver, pval < myp))
nothingsig = nrow(dplyr::filter(notsigOver, pval > myp))
chitbl = matrix(c(bothsig, selonly, diffexponly, nothingsig), nrow=2) ##make the chisq table
chitest = chisq.test(chitbl) ##run the test
  
propDiffSel = bothsig/(bothsig+diffexponly)
propNotDiffSel = selonly/(selonly+nothingsig)

par(mar=c(8,8,2,2))
contplot = function(){
bp = barplot(c(propDiffSel, propNotDiffSel), ylim=c(0,0.1), col = "#0072B2", border="white", yaxt = "n", ylab = "Proportion under selection", cex.lab=1.5)
axis(2, las=2)
axis(1, at = bp, lab = c('Cold response','Other'), cex.axis=1.5)}

contplot()

##p value histogram
diffhist = hist(sigOver$pval, plot=FALSE)
nothist = hist(notsigOver$pval, plot=FALSE)
combinedHist = rbind(nothist$density, diffhist$density)


pbarplot = function(){bp2 = barplot(combinedHist/10, beside=T, col = c("#CC79A7",   "#F0E442"), border="white", xaxt="n", yaxt="n")
axis(2, las=2)
mtext("Proportion",side=2, cex=1.5, line=4)
axis(1, at=c(0.5,bp2[1,]+2), lab = c(0,nothist$mids+0.05))
mtext("P values", side=1, cex=1.5, line=4)}

pbarplot()


postscript("../figures/coldgenes.eps",height=10,width=10,paper="special",horizontal=FALSE,colormodel="cymk", onefile=FALSE)
par(mfrow=c(1,2))
contplot()
mtext('A', cex=2,side=3, at = 0, line=2)
pbarplot()
mtext('B', cex=2,side=3, at = -0.1, line=2)
dev.off()
```

